File: AUGUST-2025/01-08-2025/p118.java
class Solution {
    public List<List<Integer>> generate(int numRows) {

        List<List<Integer>> triangle=new ArrayList<>();
        if(numRows==0) return triangle;

        triangle.add(new ArrayList<>());
        triangle.get(0).add(1);

        for(int i=1;i<numRows;i++){
            List<Integer> prev=triangle.get(i-1);
            List<Integer> new_row=new ArrayList<>();
                        // List<Integer> new_row=new ArrayList<>();
                   new_row.add(1);


            for(int j=1;j<prev.size();j++){
                new_row.add(prev.get(j-1)+prev.get(j));

            }
            new_row.add(1);
            triangle.add(new_row);
        }
        return triangle;
        
    }
}

File: AUGUST-2025/01-08-2025/p118.php
class Solution {

    /**
     * @param Integer $numRows
     * @return Integer[][]
     */
    function generate($numRows) {
        $tr = [];
        for ($i = 1; $i <= $numRows; $i++) {
            $perem = [];
            for ($j = 0; $j < $i; $j++) {
                if ($j == 0 || $j == $i - 1) $perem[] = 1;
                else {
                    $perem[] = $tr[$i - 2][$j] + $tr[$i - 2][$j - 1];
                }
            }
            $tr[] = $perem;
        }
        return $tr;
    }
}